### PR TITLE
[Dynamo] Fix alias issue with respect to wrapped numbers (#124731)

### PR DESCRIPTION
**Pull request by guilhermeleobas**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #124774

This PR fixes an issue presented when calling `aten.alias(int)` raises a TypeError.

```python
import torch
import torch.autograd.forward_ad as fwAD

def f(x):
    return 4312491 * x

device = "cpu"

with torch._subclasses.fake_tensor.FakeTensorMode():
    with fwAD.dual_level():
        x = torch.randn(3, device=device)
        y = torch.ones_like(x)
        dual = fwAD.make_dual(x, y)
        f(dual)
```

The test case above illustrates this bug.
1) `4312491` turns into a tensor that is a wrapped number
2) Forward mode AD calls `aten::alias` internally
3) The wrapped number (`4312491`) becomes a python integer
4) `aten.alias(int)` raises a `TypeError`

cc @ezyang @albanD @gqchen @pearu @nikitaved @soulitzer @Lezcano @Varal7

### CODE REVIEW

File: torch/csrc/autograd/FunctionsManual.cpp [hunk 1]
```diff
@@ -73,6 +73,9 @@ Tensor toNonOptFwGrad(const c10::optional<Tensor>& t) {
 }
 
 Tensor toNonOptPrimal(const c10::optional<Tensor>& t) {
+  // if (t.has_value()) {
+  //   std::cout << "toNonOptPrimal: " << std::endl;
+  // }
```
Comment (zou3519) [inline torch/csrc/autograd/FunctionsManual.cpp:78]: @guilhermeleobas the fix is probably:
- toNonOptPrimal should just return t if t is a wrapped number
- when functorch creates a TensorWrapper, if the thing it is wrapping is a wrapped number, then the TensorWrapper should also be a wrapped number

File: torch/csrc/utils/python_arg_parser.cpp [hunk 2]
```diff
@@ -95,7 +95,8 @@ bool should_allow_numbers_as_tensors(const std::string& name) {
       "subtract",     "subtract_",     "subtract_out", // alias of sub
       "true_divide",  "true_divide_",  "true_divide_out",
       "to",           "_to_copy",      "copy_",
-      "floor_divide", "floor_divide_", "floor_divide_out"};
+      "floor_divide", "floor_divide_", "floor_divide_out",
+      "alias"};
```
Comment (guilhermeleobas) [inline torch/csrc/utils/python_arg_parser.cpp:99]: Just adding `alias` to this list, fixes the issue. Would you be ok with that, @zou3519?

Comment (zou3519) [reply torch/csrc/utils/python_arg_parser.cpp:99]: I don't think this is the right fix, see my comment above

File: torch/csrc/autograd/FunctionsManual.cpp (hunk 1 above)
Comment (guilhermeleobas) [reply torch/csrc/autograd/FunctionsManual.cpp]: Just pushed your suggestion.

**Review by albanD — APPROVED** [review]
Sounds ok to me.

**Review by zou3519 — APPROVED** [review]
nice!

**Pull request closed without merging**